### DISCUSSION TITLE
Get Solution Version

### DISCUSSION THREAD

**Discussion by Steve-Winslade-Ceox** [microsoft/powerplatform-build-tools#385] [Ideas]
Hi,

Could there be a new action added to get the solution version for a particular solution, which would allow us to skip solution import if the solution version with the same number exists?

Thanks,

Steve

**Comment by petrochuk** [comment]
This was most upvoted one: ["smarter" solution import task](https://github.com/microsoft/powerplatform-build-tools/discussions/200). Next version of pac cli will have new flag `--skip-lower-version` added to solution import

**Reply by Steve-Winslade-Ceox** [reply]
I see the new version of the pac util has gone live which should have this in there now; is there an ETA on when this wil be available via the build tools?

**Comment by petrochuk** [comment]
You should be able to see "Skip lower version" in the latest version of build tools: ![image](https://github.com/microsoft/powerplatform-build-tools/assets/30735471/94f48fb7-9d86-4026-9fc2-b40619c8f5d4)

**Comment by Steve-Winslade-Ceox** [comment]
I can see the new tag has been added to the UI, but it does not seem to be taking effect:

[ 'Calling pac cli inputs: solution import --path D:\\a\\1\\Solutions\\AutomatedDeploymentExample_Managed.zip --async true --import-as-holding false --force-overwrite false --publish-changes false --skip-dependency-check false --convert-to-managed false --max-async-wait-time 60 --activate-plugins true --skip-lower-version true --settings-file D:\\a\\1\\SolutionParameterFiles\\AutomatedDeploymentExample.UAT.json' ]
[ 'Connected to... PEM Pro Development' ]
[ 'Connected as ***' ]
[ '' ]
[ 'Solution Importing...' ]
[ '' ]
[ 'Waiting for asynchronous operation 30d4856b-3d2b-ee11-9965-002248c6f6b4 to complete with timeout of 01:00:00' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:00 and 0.00% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:04 and 0.11% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:08 and 0.23% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:12 and 0.34% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:16 and 0.45% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:20 and 0.56% complete' ]
[ '' ]
[ 'Processing asynchronous operation... execution time: 00:00:24 and 0.68% complete' ]
[ '' ]
[ 'Asynchronous operation 30d4856b-3d2b-ee11-9965-002248c6f6b4 completed successfully within 00:00:28.4211772' ]
[ '' ]
[ 'Solution Imported successfully. Import ID: 30d4856b-3d2b-ee11-9965-002248c6f6b4' ]


The solution being imported has the same version number as the one in the environment. I've tried this as both Upgrade and Update approaches and both show the same behaviour.

Also, if/when it works, would the step report as skipped if it doesn't run as the solution is already up to date, so that subsequent steps (e.g. apply solution upgrade can be skipped)

Thanks!

**Comment by jcoxdco** [comment]
+1 on the suggestion that either the step reports as skipped or perhaps an output variable is set indicating the action taken by the import task. Doesn't seem there's a way (or a straight forward way) to not run a subsequent Upgrade task based on the import being the same or higher version.

**Comment by YannickRe** [comment]
Seems like this one, #494 , and #501 all highlight the same (or at least a similar problem).

**Comment by parvezghumra** [comment]
`pac solution online-version` now supports retrieving solution version in v2.5.1 of PAC CLI
